enhancements/testing/ephemeral-agent-installer-e2e-tests.md

```diff
@@ -0,0 +1,329 @@
+need to have defaults and be aware of those environment variables.
+
+#### Build
+
+The scripts/automation for building the agent-based installer should be inside the repository
+where the installer is being worked on and not dev-scripts. We will reference the built 
+container images and ISO in dev-scripts.
+
+#### Deployment
+
+Deploying the agent-based installer will be heavily linked to the environment set up. This
+logic will land in dev-scripts alongside the pre environment setup steps.
+
+#### Tests
+
+For the actual test scenarios themselves we should use Ginkgo/Gomega. This is 
+similar to the rest of Openshift. Well most of (if not all) the product is written in Golang
+so it makes sense to use a Golang framework for these.
+
+There is a repository created as a preliminary place to put these. Plan to use a similar
+structure to openshift-tests in this repository.
+
+We will be including existing e2e tests that have been created to validate the OCP cluster
+installed by the new agent-based installer.
+
+#### Teardown
+
+Currently there is a `make clean` flow for starting new locally. We will need to add similar
+functionally for the flows that will be used for the agent-based installer
+
+### Open Questions [optional]
+
+### Test Plan
+
+* e2e tests for the three install scenarios of cluster 0 (SNO, Compact, HA)
+* Run Openshift compliance tests against these installed clusters
+
+### Graduation Criteria
+
+#### Dev Preview -> Tech Preview
+
+* Ability to utilize the e2e framework locality to aid in development
+* Openshift compliance tests are run against the installed clusters
+* The e2e framework is on boarded into Openshift CI for the team's use
+
+#### Tech Preview -> GA
+
+* Add upgrade/downgrade e2e tests
+
+#### Removing a deprecated feature
+
+### Upgrade / Downgrade Strategy
+
+### Version Skew Strategy
+
+* Currently the dev-scripts repository is not versioned. This will stay the same.
+
+* However for the e2e test scenarios themselves it makes sense to version these.
+Since the ephemeral agent based installer is targeted to be released with OCP it would 
+be easiest to version the e2e framework the same as OCP x.y. This is the strategy
+that openshift-tests takes.
+
+### Operational Aspects of API Extensions
+
+#### Failure Modes
+
+#### Support Procedures
+
+## Implementation History
+
+## Drawbacks
+
+The biggest argument against it is that by creating a new framework we don't
+inherit the work that was done previously in validating Assisted Installer.
+
+## Alternatives
+
+~~Alternatively we can re-factor one of the existing frameworks to include this
+new project.~~
+
+Based on the feedback for this enhancement we will be extending parts of dev-scripts
+to include this new project. However, as it is outlined above not all components of
+this e2e framework will land in the dev-scripts repository. Specifically the components
+for the Build and Test phases will land in other repositories.
+
```

Comment:
There were basically 3 options originally, right? Modify dev-scripts, modify test-infra, or create something completely new. The 2 options that were rejected should be described here briefly, including the reasons for not choosing them.